### ISSUE TITLE
Model::asModel() only works if new Model has all fields of current

### ISSUE THREAD

**Issue by PhilippGrashoff** [atk4/data#457]
Hi there,

currently asModel() can only be used to copy values from class A to class B if B has all fields of class A.
Example:
```
Class A
$this->addFields([
    ['field1'],
    ['field2'],
    ['field3'],
]);

Class B
$this->addFields([
    ['field1'],
    ['field2'],
    ['field3'],
    ['field4'],
]);

Class C
$this->addFields([
    ['field1'],
    ['field2'],
]);


//works
$new_b = $someInstanceOfA->asModel(B::class);
//doenst because C does not have field3
$new_c = $someInstanceOfA->asModel(C::class);
```

I think it should be possible to also use asModel() with a class that has fewer fields than the initial one.

Also, there is $options parameter which is currently unused, and a comment about dirty fields.

I can update asModel to fit this issue and the dirty fields if you like.

**Comment by mvorisek** [comment]
> I think it should be possible to also use asModel() with a class that has fewer fields than the initial one.

the new model should be instanceof this, otherwise throw. If model has same fields, it does not mean anything (imagine Currency and Country models with name and iso_code fields)

**Comment by DarkSide666** [comment]
It's not always `instanceof`. It can also be that both models share same parent model. Imagine Invoice and child classes DraftInvoice, PaidInvoice. We should be able to do DraftInvoice->asModel(PaidInvoice)

**Comment by mvorisek** [comment]
Yes, then only fields from parent model should be copied... But I have no use case for this method.

**Comment by DarkSide666** [comment]
I have used it (saveAs actually) to migrate _Quote_ record from quote table to _Invoice_ table for example `Quote->saveAs(Invoice)`.
Quote and Invoice classes both are quite similar and share same parent class - Document.

**Comment by mvorisek** [comment]
But you can always use foreach and do it manually. Model methods should work always. If they can works only partly, they are wrongly designed (like expect that field is for the same if it has the same name etc.).

**Comment by mvorisek** [comment]
`asModel` was removed completely :)
